### ISSUE TITLE
"No such type" error in RC5

### ISSUE THREAD

**Issue by jclark** [ballerina-platform/ballerina-lang#30816]
I tried switching everything to a single package.  I put it in this branch

https://github.com/ballerina-platform/nballerina/tree/james-single-package

It compiles OK with (bal build --skip-tests), but when I run e.g. using

```
bal run -- --showTypes tests\data\basic.bal
```

I get:

```
 No such type: Package: ballerina/lang$0046__internal/0_1_0/$_init, TypeName: $type$$anonType$_12, Shape: record {| int value; |}
```

Not sure if this is the same as #30814.

**Comment by jclark** [comment]
When I run the tests I get some location info:

```
No such type: Package: ballerina/lang$0046__internal/0_1_0/$_init, TypeName: $type$$anonType$_12, Shape: record {| int value; |}

                        at wso2.nballerina.types.0_1_0:listInhabited(list.bal:222)
                        wso2.nballerina.types.0_1_0:listFormulaIsEmpty(list.bal:171)
                        wso2.nballerina.types.0_1_0:listFormulaIsEmpty$lambda0$(list.bal:112)
                        wso2.nballerina.types.0_1_0:bddEvery(common.bal:51)
                        wso2.nballerina.types.0_1_0:bddEvery(common.bal:56)
                        wso2.nballerina.types.0_1_0:bddEvery(common.bal:54)
                        wso2.nballerina.types.0_1_0:listSubtypeIsEmpty(list.bal:112)
                        wso2.nballerina.types.0_1_0:listRoSubtypeIsEmpty(list.bal:89)
                        wso2.nballerina.types.0_1_0:isEmpty(core.bal:567)
                        wso2.nballerina.types.0_1_0:isSubtype(core.bal:576)
                        wso2.nballerina.types.0_1_0.tests.coretest:tupleTest1(tests/coretest.bal:113)
```

**Comment by jclark** [comment]
This same code worked in slalpha5, except for change of module names.

**Comment by hasithaa** [comment]
Fixed with #30836
